services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/doi/DoiManagerImpl.java

```diff
@@ -240,10 +242,17 @@ String generateLocationRequestUrl(DoiAssociation association) {
 		final String PERSISTENT_REPOSITORY_ENDPOINT = "https://repo-" + stack + "." + stack + ".sagebase.org/repo/v1";
 		
 		String request = PERSISTENT_REPOSITORY_ENDPOINT + LOCATE_RESOURCE_PATH;
-		
+
+		String encodedId;
+		try {
+			encodedId = URLEncoder.encode(association.getObjectId(), StandardCharsets.UTF_8.toString());
```

Comment:
We might have to do the same in the /locate API when we redirect to an external portal URL